lib/matplotlib/cbook/deprecation.py

```diff
@@ -4,6 +4,41 @@
 import warnings
 
 
+# has to be here to prevent circular import, since it's used by `deprecated`
+class _classproperty:
```

Comment:
This movement and the changes in `MatplotlibDeprecationWarning` seem unrelated. While needed here, they are too complex to be burried in a large change. I'd prefer them at least in a seprate commit, better in a dedicated PR. (My brain is not large enough to think this though while keeping all the style changes in mind).

Reply:
I think you can just do a local import within the implementation of `cbook.deprecated`?

Reply:
Yup, this should have been separate from the start. @anntzer I'll make that change in #18817 if that's the preferred way to fix this problem.